pkg/plugin/processor/builtin/unwrap/unwrap_kafkaconnect.go

```diff
@@ -0,0 +1,176 @@
+
+	fieldRefRes sdk.ReferenceResolver
+}
+
+func NewKafkaConnectProcessor(log.CtxLogger) *kafkaConnectProcessor {
+	return &kafkaConnectProcessor{}
+}
+
+func (u *kafkaConnectProcessor) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:    "unwrap.kafkaconnect",
+		Summary: "Unwraps a Kafka Connect record from an OpenCDC record.",
+		Description: `This processor unwraps a Kafka Connect record from the input OpenCDC record.
+
+The input record's payload is replaced with the Kafka Connect record.
+
+This is useful in cases where Conduit acts as an intermediary between a Debezium source and a Debezium destination. 
+In such cases, the Debezium record is set as the OpenCDC record's payload, and needs to be unwrapped for further usage.`,
+		Version:    "v0.1.0",
+		Author:     "Meroxa, Inc.",
+		Parameters: kafkaConnectConfig{}.Parameters(),
+	}, nil
+}
+
+func (u *kafkaConnectProcessor) Configure(ctx context.Context, m map[string]string) error {
+	cfg := kafkaConnectConfig{}
+	err := sdk.ParseConfig(ctx, m, &cfg, cfg.Parameters())
+	if err != nil {
+		return err
+	}
+
+	rr, err := sdk.NewReferenceResolver(cfg.Field)
+	if err != nil {
+		return cerrors.Errorf("invalid reference: %w", err)
+	}
+
+	u.fieldRefRes = rr
+	return nil
+}
+
+func (u *kafkaConnectProcessor) Open(context.Context) error {
+	return nil
+}
+
+func (u *kafkaConnectProcessor) Process(_ context.Context, records []opencdc.Record) []sdk.ProcessedRecord {
+	out := make([]sdk.ProcessedRecord, 0, len(records))
+	for _, rec := range records {
+		proc, err := u.processRecord(rec)
+		if err != nil {
+			return append(out, sdk.ErrorRecord{Error: err})
+		}
+		out = append(out, proc)
+	}
+
+	return out
+}
+
+func (u *kafkaConnectProcessor) processRecord(rec opencdc.Record) (sdk.ProcessedRecord, error) {
+	// record must be structured
+	ref, err := u.fieldRefRes.Resolve(&rec)
+	if err != nil {
+		return nil, cerrors.Errorf("failed resolving reference: %w", err)
+	}
+
+	var kc opencdc.StructuredData
+	switch d := ref.Get().(type) {
+	case opencdc.StructuredData:
+		kc = d
+	case map[string]any:
+		kc = d
+	case opencdc.RawData:
+		err := json.Unmarshal(d.Bytes(), &kc)
+		if err != nil {
+			return nil, cerrors.Errorf("failed unmarshalling JSON from raw data: %w", err)
+		}
+	case string:
+		err := json.Unmarshal([]byte(d), &kc)
+		if err != nil {
+			return nil, cerrors.Errorf("failed unmarshalling JSON from string: %w", err)
+		}
+	default:
+		return nil, cerrors.Errorf("unexpected data type %T (only structured data is supported)", ref.Get())
+	}
+
+	// get payload
+	structPayload, ok := kc["payload"].(map[string]any)
+	if !ok {
+		return nil, cerrors.New("referenced record field doesn't contain a payload field")
+	}
+
+	return sdk.SingleRecord{
+		Key:      u.unwrapKey(rec.Key),
+		Position: rec.Position,
+		Metadata: rec.Metadata,
+		Payload: opencdc.Change{
+			After: opencdc.StructuredData(structPayload),
+		},
+		Operation: opencdc.OperationSnapshot,
```

Comment:
@lovromazgon @maha-hajja I wanted to clarify this. IIUC, given that these records come from Kafka, where we don't have any updates or deletes, the operation can only be `Create` or `Snapshot`. Is there a special reason why we chose `Snapshot` here? Taken from: https://github.com/ConduitIO/conduit/blob/main/pkg/processor/procbuiltin/unwrap.go#L336

Reply:
Hmm, that should probably be `OperationCreate` if you ask me.

Reply:
I don't think there was a specific reason no, I agree with changing it to `OperationCreate`

Reply:
Thanks!